src/Analysis/Engine/Impl/ProjectEntry.cs

```diff
@@ -157,9 +157,13 @@ private class ActivePythonParse : IPythonParse {
 
         internal void SetCompleteAnalysis() {
             lock (this) {
-                _analysisTcs.TrySetResult(Analysis);
+                if (AnalysisVersion == _expectedParse) {
```

Comment:
this can be simplified a bit:
```cs
lock (this) {
    if (AnalysisVersion != _expectedParse) {
        return;
    }
    _analysisTcs.TrySetResult(Analysis);
}
RaiseNewAnalysis(); 
```